File: app/src/main/java/com/spidroid/starry/ui/messages/PollDialog.java
package com.spidroid.starry.ui.messages;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Toast;
import com.google.android.material.bottomsheet.BottomSheetDialog;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;
import com.spidroid.starry.R;
import java.util.ArrayList;
import java.util.List;

public class PollDialog extends BottomSheetDialogFragment {

  public interface OnPollCreatedListener {
    void onPollCreated(String question, List<String> options);
  }

  private OnPollCreatedListener listener;
  private LinearLayout optionsContainer;
  private int optionCount = 2;

  public void setOnPollCreatedListener(OnPollCreatedListener listener) {
    this.listener = listener;
  }

  @Override
  public View onCreateView(
      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.dialog_poll, container, false);
    setupPollCreation(view);
    return view;
  }

  private void setupPollCreation(View view) {
    optionsContainer = view.findViewById(R.id.options_container);
    EditText pollQuestion = view.findViewById(R.id.poll_question);
    Button addOption = view.findViewById(R.id.add_option);
    Button btnCancel = view.findViewById(R.id.btn_cancel);
    Button btnCreate = view.findViewById(R.id.btn_create);

    addOption.setOnClickListener(v -> addNewOptionField());
    btnCancel.setOnClickListener(v -> dismiss());
    btnCreate.setOnClickListener(v -> createPoll(pollQuestion));
  }

  private void addNewOptionField() {
    if (optionCount >= 6) return; // Limit to 6 options

    TextInputLayout inputLayout = new TextInputLayout(requireContext());
    inputLayout.setLayoutParams(
        new LinearLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
    inputLayout.setHint("Option " + (optionCount + 1));

    TextInputEditText editText = new TextInputEditText(inputLayout.getContext());
    editText.setLayoutParams(
        new LinearLayout.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
    inputLayout.addView(editText);

    optionsContainer.addView(inputLayout);
    optionCount++;
  }

  private void createPoll(EditText questionEditText) {
    String question = questionEditText.getText().toString().trim();
    List<String> options = new ArrayList<>();

    // Collect options from all input fields
    for (int i = 0; i < optionsContainer.getChildCount(); i++) {
      TextInputLayout inputLayout = (TextInputLayout) optionsContainer.getChildAt(i);
      EditText editText = (EditText) inputLayout.getChildAt(0);
      String option = editText.getText().toString().trim();
      if (!option.isEmpty()) {
        options.add(option);
      }
    }

    if (question.isEmpty()) {
      questionEditText.setError("Please enter a question");
      return;
    }

    if (options.size() < 2) {
      Toast.makeText(requireContext(), "At least 2 options required", Toast.LENGTH_SHORT).show();
      return;
    }

    if (listener != null) {
      listener.onPollCreated(question, options);
    }
    dismiss();
  }
}

File: app/src/main/java/com/spidroid/starry/ui/messages/PollDialog.kt
package com.spidroid.starry.ui.messages

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.Toast
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout
import com.spidroid.starry.R
import com.spidroid.starry.models.ChatMessage
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class PollDialog : BottomSheetDialogFragment() {

    // واجهة لإعادة بيانات الاستطلاع إلى الـ Activity
    interface OnPollCreatedListener {
        fun onPollCreated(poll: ChatMessage.Poll)
    }

    private var listener: OnPollCreatedListener? = null
    private lateinit var optionsContainer: LinearLayout
    private lateinit var pollQuestionEditText: TextInputEditText
    private val optionEditTexts = mutableListOf<TextInputEditText>()

    fun setOnPollCreatedListener(listener: OnPollCreatedListener) {
        this.listener = listener
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        val view = inflater.inflate(R.layout.dialog_poll, container, false)
        setupViews(view)
        return view
    }

    private fun setupViews(view: View) {
        optionsContainer = view.findViewById(R.id.options_container)
        pollQuestionEditText = view.findViewById(R.id.poll_question)

        // إضافة حقلي الخيارات الأوليين إلى القائمة
        optionEditTexts.add(view.findViewById(R.id.option_1))
        optionEditTexts.add(view.findViewById(R.id.option_2))

        val addOptionButton: Button = view.findViewById(R.id.add_option)
        val btnCancel: Button = view.findViewById(R.id.btn_cancel)
        val btnCreate: Button = view.findViewById(R.id.btn_create)

        addOptionButton.setOnClickListener { addNewOptionField() }
        btnCancel.setOnClickListener { dismiss() }
        btnCreate.setOnClickListener { createPoll() }
    }

    private fun addNewOptionField() {
        if (optionEditTexts.size >= 4) { // الحد الأقصى 4 خيارات
            Toast.makeText(context, "Maximum of 4 options allowed.", Toast.LENGTH_SHORT).show()
            return
        }

        // إنشاء حقل إدخال جديد ديناميكيًا
        val newOptionLayout = LayoutInflater.from(context).inflate(R.layout.item_poll_option_input, optionsContainer, false) as TextInputLayout
        val newOptionEditText = newOptionLayout.findViewById<TextInputEditText>(R.id.poll_option_input)
        newOptionLayout.hint = "Option ${optionEditTexts.size + 1}"

        optionsContainer.addView(newOptionLayout)
        optionEditTexts.add(newOptionEditText)
    }

    private fun createPoll() {
        val question = pollQuestionEditText.text.toString().trim()
        if (question.isBlank()) {
            pollQuestionEditText.error = "Question cannot be empty"
            return
        }

        val options = optionEditTexts
            .map { it.text.toString().trim() }
            .filter { it.isNotBlank() }

        if (options.size < 2) {
            Toast.makeText(context, "At least 2 options are required.", Toast.LENGTH_SHORT).show()
            return
        }

        val pollOptions = options.map { ChatMessage.PollOption(text = it) }
        val poll = ChatMessage.Poll(question = question, options = pollOptions.toMutableList())

        listener?.onPollCreated(poll)
        dismiss()
    }
}